test/other/test_emmalloc_in_addition.c

```diff
@@ -0,0 +1,36 @@
+#include <assert.h>
+#include <emscripten/console.h>
+#include <emscripten/emmalloc.h>
+
+int main() {
+  // Verify we can call both malloc and emmalloc_malloc, and that those are
+  // different functions, unless TEST_EMMALLOC_IS_MALLOC is set (in that case,
+  // emmalloc is malloc because we let emmalloc define the standard exports like
+  // malloc).
+
+  // We have allocated nothing so far, but there may be some initial allocation
+  // from startup.
+  size_t initial = emmalloc_dynamic_heap_size();
+  emscripten_console_logf("initial: %zu\n", initial);
+
+  const size_t ONE_MB = 1024 * 1024;
+  void* one = malloc(ONE_MB);
+  assert(one);
+#ifndef TEST_EMMALLOC_IS_MALLOC
+  // We have allocated using malloc, but not emmalloc, so emmalloc reports no
+  // change in usage.
+  assert(emmalloc_dynamic_heap_size() == initial);
+#else
+  // malloc == emmalloc_malloc, so emmalloc will report additional usage (of the
```

Comment:
You could maybe simplify this test by simply asserting that `malloc == emmalloc_malloc` as the command says?    The you wouldn't need the comment to describe the code as it would more self explanatory?

Reply:
By taking the address, you mean? I wasn't sure that is entirely reliable in general. In particular I'd worry about there being some kind of trampoline but it still going to the same target. (With aliases that wouldn't happen, but we might refactor to use another method.)

Reply:
If we use another method I think its fine to update this test at that point... but that other method would seem like a bad idea since it would incur a cost.

Reply:
I think there would be a risk that we could forget to update this test... it would be a needle in the haystack of the testsuite, that is, we'd need to somehow realize it needs to be updated then. Most likely we'd forget and it could start to pass incorrectly.

Reply:
But surely the tests would fail because the assert "malloc == emmalloc" would fail right away, no?

Reply:
In other words, I think it's safer to test this by checking the functionality - that `emmalloc` actually runs, or does not -  and not an indirect property that is connected to that atm, if that makes sense.

Reply:
Fair enough.   This lgtm.

Reply:
I realize you're right that `malloc == emmalloc` would fail one way at least, if we check both ways (as this PR currently does, so one path checks `==` and one `!=`). So my argument isn't as strong as I thought. Still, I think this is a pretty simple way to check the underlying functionality so I do feel it is safer.

Reply:
I tried to add `emmalloc == malloc` testing as you suggested, because I thought it would be good to have both, but actually that fails atm, it turns out! We *do* have a trampoline:

https://github.com/emscripten-core/emscripten/blob/4608c2eb202652f00115ba71232a89010a4553ac/system/lib/emmalloc.c#L836-L839

So in a debug build the two are obviously different. It turns out that even in a release build they are, because LLVM allocates two function pointer addresses, one for each. Optimizations (inlining + merging similar functions) end up making them identical, but there are still two function pointers, and the table ends up with two slots with the same function name in it...

It might be worth using an alias rather than a trampoline here, but I'm not sure if there was a reason not to do that. Anyhow, let's leave that separate from this PR, so I'll land this.

Reply:
I can't get `emmalloc == malloc` testing to work in a followup.

I fixed the code to not have trampolines, and the function pointers are identical, and it works in a debug build, but the LLVM optimizer assumes they are not equal all the time. I seem to remember something about that being undefined behavior, that in some architectures function pointers behave very oddly... so I guess it's that.